### PR TITLE
fix: update notification template variables for `user_given_top_reader`

### PR DESCRIPTION
**Pull request by omBratteng**

### CODE REVIEW

File: src/workers/newNotificationV2Mail.ts [hunk 1]
```diff
@@ -717,8 +717,18 @@ const notificationToTemplateData: Record<NotificationType, TemplateDataFunc> = {
...
 
+    const keyword = await userTopReader.keyword;
+    if (!keyword) {
+      throw new Error('Keyword not found');
+    }
+
     return {
-      topReader: userTopReader.image,
+      image: userTopReader.image,
+      issuedAt: new Date(userTopReader.issuedAt).toLocaleString('en-US', {
```
Comment (capJavert) [inline src/workers/newNotificationV2Mail.ts:727]: `formatMailDate`?

Comment (omBratteng) [reply src/workers/newNotificationV2Mail.ts:727]: That would format `2024-10-31T00:00:00.000Z` as `Oct 31, 2024`. We want it to be `October 2024` to match format in front-end for the badges.

Comment (omBratteng) [reply src/workers/newNotificationV2Mail.ts:727]: image

Comment (capJavert) [reply src/workers/newNotificationV2Mail.ts:727]: I thought they can format inside the templates

Comment (omBratteng) [reply src/workers/newNotificationV2Mail.ts:727]: Probably, it's just not done in this template. I could take a look

Comment (omBratteng) [reply src/workers/newNotificationV2Mail.ts:727]: It's possible. Will use `formatMailDate`

**Review by capJavert — APPROVED** [review]

**Pull request merged**